kurtosis-devnet/tests/eigenda/failover_test.go

```diff
@@ -0,0 +1,409 @@
+	// We'll use standard HTTP for GraphQL as it's not directly supported by the rpc package
+	query := fmt.Sprintf(`
+	{
+		"query": "query txInfo { blocks(from:%v) { transactions { to { address } inputData block { number } } } }"
+	}`, blockNum)
+
+	// Make GraphQL request
+	req, err := http.NewRequest("POST", gethL1Endpoint+"/graphql", strings.NewReader(query))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	httpClient := &http.Client{}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// Parse the response
+	type GraphQLResponse struct {
+		Data struct {
+			Blocks []struct {
+				Transactions []struct {
+					To struct {
+						Address string `json:"address"`
+					} `json:"to"`
+					InputData string `json:"inputData"`
+					Block     struct {
+						// we use HexUint64 to properly parse the hex strings returned
+						Number HexUint64 `json:"number"`
+					} `json:"block"`
+				} `json:"transactions"`
+			} `json:"blocks"`
+		} `json:"data"`
+	}
+	var graphQLResp GraphQLResponse
+	if err := json.NewDecoder(resp.Body).Decode(&graphQLResp); err != nil {
+		return nil, err
+	}
+	if len(graphQLResp.Data.Blocks) == 0 {
+		// Assume that this is a graphQL query error, that would have returned something like
+		// "errors": [
+		// 	{
+		// 	  "message": "syntax error: unexpected \"\", expecting Ident",
+		// 	}
+		// ]
+		// TODO: prob should just switch to a proper graphql client that can handle these properly
+		return nil, fmt.Errorf("no blocks returned in GraphQL response")
+	}
+
+	// Filter transactions to the batcher address
+	var batcherTxs []BatcherTx
+	for _, block := range graphQLResp.Data.Blocks {
+		for _, tx := range block.Transactions {
+			if strings.EqualFold(tx.To.Address, batchInbox) {
+				var daLayer DALayer
+				if strings.HasPrefix(tx.InputData, eigenDACommitmentPrefix) {
+					daLayer = DALayerEigenDA
+				} else if strings.HasPrefix(tx.InputData, ethDACommitmentPrefix) {
+					daLayer = DALayerEth
+				} else {
+					return nil, fmt.Errorf("unknown commitment prefix: %s", tx.InputData)
+				}
+				batcherTxs = append(batcherTxs, BatcherTx{
+					commitment: tx.InputData,
+					daLayer:    daLayer,
+					block:      uint64(tx.Block.Number),
+				})
+			}
+		}
+	}
+
+	return batcherTxs, nil
+}
+
+// Localhost endpoints for the different services in the enclave
+// that we need to interact with.
+type EnclaveServiceEndpoints struct {
+	OpNodeEndpoint       string `kurtosis:"op-cl-1-op-node-op-geth-op-kurtosis,http"`
+	GethL1Endpoint       string `kurtosis:"el-1-geth-teku,rpc"`
+	EigendaProxyEndpoint string `kurtosis:"da-server-op-kurtosis,http"`
+	// Adding new endpoints is as simple as adding a new field with a kurtosis tag
+	// NewServiceEndpoint   string `kurtosis:"new-service-name,port-name"`
+}
+
+func getEndpointsFromKurtosis(enclaveCtx *enclaves.EnclaveContext) (*EnclaveServiceEndpoints, error) {
```

Comment:
A doc here would be helpful. It looks like this method replaces the service name in the `EnclaveServiceEndpoints` struct with a valid localhost endpoint?